numpy/lib/_function_base_impl.py

```diff
@@ -3827,7 +3827,8 @@ def sinc(x):
 
     """
     x = np.asanyarray(x)
-    y = pi * where(x == 0, 1.0e-20, x)
+    x = x.astype(np.float64) if np.isdtype(x.dtype, 'integral') else x
```

Comment:
`sinc` should handle boolean inputs too:
```suggestion
    x = x.astype(np.float64) if np.isdtype(x.dtype, ('bool', 'integral')) else x
```

Reply:
oops, lost this commit during rebase